app/src/main/java/nmd/nomad/api/ServiceGenerator.java

```diff
@@ -0,0 +1,21 @@
+package nmd.nomad.api;
+
+import com.squareup.okhttp.OkHttpClient;
+
+import retrofit.RestAdapter;
+import retrofit.client.OkClient;
+
+/**
+ * Created by nicolasiensen on 1/28/15.
+ */
+public class ServiceGenerator {
+    public static <S> S createService(Class<S> serviceClass, String baseUrl) {
+        RestAdapter.Builder builder = new RestAdapter.Builder()
+                .setEndpoint(baseUrl)
+                .setClient(new OkClient(new OkHttpClient()));
+
+        RestAdapter adapter = builder.build();
+
+        return adapter.create(serviceClass);
```

Comment:
It is recommended for Retrofit that this is only instantiated once and stored as a singleton.